dbclient/WorkspaceClient.py

```diff
@@ -253,12 +257,13 @@ def download_notebooks(self, ws_log_file='user_workspace.log', ws_dir='artifacts
             # pull the path from the data to download the individual notebook contents
             for notebook_data in fp:
                 notebook_path = json.loads(notebook_data).get('path', None).rstrip('\n')
-                dl_resp = self.download_notebook_helper(notebook_path, checkpoint_notebook_set, export_dir=self.get_export_dir() + ws_dir)
+                dl_resp = self.download_notebook_helper(notebook_path, checkpoint_notebook_set, notebook_error_logger,
+                                                        export_dir=self.get_export_dir() + ws_dir)
                 if 'error' not in dl_resp:
                     num_notebooks += 1
         return num_notebooks
 
-    def download_notebook_helper(self, notebook_path, checkpoint_notebook_set, export_dir='artifacts/'):
+    def download_notebook_helper(self, notebook_path, checkpoint_notebook_set, error_logger, export_dir='artifacts/'):
```

Comment:
I think it's a common pattern to define logger as glob variable to void passing it around.

Reply:
error_logger is per object type, and each Client deals with multiple objects. If we want to define it as global variables, we need to define multiple loggers (by action_type & object_type). Therefore I preferred getting the logger where needed and passing it around